server/tools/jgroups.sh

```diff
@@ -18,3 +18,9 @@ if [ -n "$JGROUPS_DISCOVERY_PROTOCOL" ]; then
        $JBOSS_HOME/bin/jboss-cli.sh --file="/opt/jboss/tools/cli/jgroups/discovery/default.cli" >& /dev/null
     fi
 fi
+
+# How many owners / replicas should our distributed caches have. If <2 any node that is removed from the cluster will cause a data-loss!
+if [ -n "$JGROUPS_CACHE_OWNERS" ]; then
```

Comment:
This actually should be Infinispan. JGroups project is about cluster communication but it is Infinispan that stores data in a distributed fashion. Perhaps it wouls be even better to name it `NUMBER_OF_CACHE_OWNERS`, @mposolda @stianst @abstractj WDYT?

Apart from that, LGTM.

Reply:
It should have a prefix, but JGROUPS is not the correct prefix. Perhaps CACHE_OWNERS_COUNT. One concern I have here is that it is a single value that sets the number of owners for everything. For example authenticationSessions shouldn't need to be replicated as it is a short lived temporary session used during authentication. So at least that cache should be configured separately from the rest.

Reply:
+1 for CACHE_OWNERS_COUNT and authenticationSessions having possibility of separate count. Will be good to document that for authenticationSessions, it is recommended to stick with 1 in most cases. Or maybe even keep this one hardcoded to 1 and document the limitation that authentication eventually needs to start from scratch in case of failover?

Reply:
> Or maybe even keep this one hardcoded to 1 and document the limitation that authentication eventually needs to start from scratch in case of failover?

@mposolda what about Kubernetes/OpenShift use case where Keycloak operates behind a Load Balancer? You never know which Keycloak instance would be used. Would your proposal work in this scenario?

Reply:
It will still work then as it's a distributed cache. Authentication sessions are short lived and updated on every request, so replicating them really isn't good.